internal/v1/handler.go

```diff
@@ -299,6 +292,26 @@ func (h *Handlers) GetComposeStatus(ctx echo.Context, composeId uuid.UUID) error
 	return ctx.JSON(http.StatusOK, status)
 }
 
+func (h *Handlers) handleNotFoundResponse(resp *http.Response) error {
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return echo.NewHTTPError(http.StatusInternalServerError, "Failed to read response body").SetInternal(err)
+	}
+	// Composes can get deleted in composer, usually when the image is expired
+	return echo.NewHTTPError(http.StatusNotFound, string(body))
+}
+
+func (h *Handlers) handleErrorResponse(ctx echo.Context, resp *http.Response) error {
+	httpError := echo.NewHTTPError(http.StatusInternalServerError, "Failed querying compose status")
```

Comment:
The error message is about `compose status` but the function name is very generic so I would suggest to either make the function name more specific (maybe `handleErrorResponseForComposeStatus`) *or* (probably better) pass something like a "errorContext" or "errorDescription" into the helper, so something like:
```go
func (h *Handlers) handleErrorResponse(ctx echo.Context, errorDescription string, resp *http.Response) error {
	httpError := echo.NewHTTPError(http.StatusInternalServerError, errorDescription)
...
```
(and the same for the ctx.Logger() below) and of course the caller now needs to pass `"Failed querying compose status"` or if a different function uses this helper the caller they can use the appropriate message.

This way the helper can be reused in all places that need to handle error responses.